sql/core/src/main/scala/org/apache/spark/sql/DataFrameImpl.scala

```diff
@@ -141,7 +141,7 @@ private[sql] class DataFrameImpl protected[sql](
   }
 
   override def orderBy(sortCol: String, sortCols: String*): DataFrame = {
-    sort(sortCol, sortCols :_*)
+    orderBy(sortCol, sortCols :_*)
```

Comment:
Can you put the real types here instead? (i.e. `sortCols: String*`) as the flow here is super confusing.  Also I'd have most versions call `sort(sortCol: String, sortCols: String*)`, where the actual work is done.